### PR TITLE
#39 Setup Cloudinary: Close

### PR DESCRIPTION
**Pull request by DejanKatovic**
### What this PR does (required):
- Create Cloudinary account and add APIs to .env file
- Create a route for profile image uploading
- Upload images to Cloudinary and returns URLs of uploaded images.

### Screenshots / Videos (required):
img1
img2

### Any information needed to test this feature (required):
- Cloudinary APIs will be shown in the slack channel.

### Any issues with the current functionality (optional):
- Profile model is not complete yet so I did not implement saving URLs to MongoDB.

### CODE REVIEW

File: server/controllers/profile.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+exports.savePhoto = asyncHandler(async (req, res, next) => {
+	const {photos} = req.files;
+
+	const uploadPromises = photos.map(photo => cloudinary.uploader.upload(photo.path));
+
+	const results = await Promise.all(uploadPromises);
+	const urls = results.map(result => result.url);
+
+	photos.forEach(photo => {
+		fs.unlinkSync(photo.path);
```
Comment (ezeilo-su) [inline server/controllers/profile.js:15]: Use proper indentation

File: server/routes/profile.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const express = require("express");
+const multer = require("multer");
+const router = express.Router();
+const protect = require("../middleware/auth");
+const { savePhoto } = require("../controllers/profile");
+
+const upload = multer({dest: './uploads'});
+
+// router.route("/photo").post(protect, savePhoto);
+router.route("/savePhoto").post(upload.fields([{name: 'photos'}]), savePhoto);
```
Comment (ezeilo-su) [inline server/routes/profile.js:10]: Good implementation. You may want to break `.post(...` to the line to make for better readability.

**Review by ezeilo-su — COMMENTED** [review]
- Good PR description message, including screenshots.
- Code is well organized.
- :+1:

File: server/routes/profile.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const express = require("express");
+const multer = require("multer");
+const router = express.Router();
+const protect = require("../middleware/auth");
+const { savePhoto } = require("../controllers/profile");
+
+const upload = multer({dest: './uploads'});
+
+// router.route("/photo").post(protect, savePhoto);
```
Comment (bonnieli) [inline server/routes/profile.js:9]: unused code?

Comment (DejanKatovic) [reply server/routes/profile.js:9]: It will be used for ticket #41.

File: server/routes/profile.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+router.route("/savePhoto")
```
Comment (rajivtitus) [inline server/routes/profile.js:10]: Please use hyphenated case for route names with multiple words :)

Eg: `edit-profile`

File: server/routes/profile.js (hunk 3 above)
Comment (rajivtitus) [reply server/routes/profile.js:9]: It is a good practice to add a 'TODO comment' with a description along with commented code :)

File: server/routes/profile.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+const express = require("express");
+const multer = require("multer");
+const router = express.Router();
+const protect = require("../middleware/auth");
+const { savePhoto } = require("../controllers/profile");
+
+const upload = multer({dest: './uploads'});
```
Comment (rajivtitus) [inline server/routes/profile.js:7]: As an additional step, maybe add file type validation?

Ref: https://www.positronx.io/multer-file-type-validation-tutorial-with-example/

**Review by rajivtitus — APPROVED** [review]
Good work with this! Just a couple of minor comments to address before merging.
Thanks :)

**Pull request merged**